test/ERC7432.spec.ts

```diff
@@ -201,17 +204,70 @@ describe('Nfts Roles', () => {
         const returnedData = await nftRoles.roleData(role, roleCreator.address, userOne.address, AddressZero, tokenId)
         expect(returnedData).to.equal(customData)
 
-        const returnedExpirationDate = await nftRoles.roleExpirationDate(role, roleCreator.address, userOne.address, AddressZero, tokenId)
+        const returnedExpirationDate = await nftRoles.roleExpirationDate(
+          role,
+          roleCreator.address,
+          userOne.address,
+          AddressZero,
+          tokenId,
+        )
         expect(returnedExpirationDate).to.equal(expirationDate)
       })
-    })
+      it('should decode role custom data with metadata', async () => {
+        const nftMetadata = {
+          name: 'Nft name',
+          description: 'Nft description',
+          image: 'https://example.com/image.png',
+          roles: [
+            {
+              name: 'Role Name',
+              description: 'User of the Nft',
+              inputs: [
+                {
+                  name: 'user',
+                  type: 'address',
+                  description: 'User address',
```

Comment:
Maybe use the real estate example listed on the EIP? With two roles, one unique and the other not